packages/front-end/components/AutoGenerateMetricsModal.tsx

```diff
@@ -0,0 +1,299 @@
+import { useState, useEffect, useCallback } from "react";
+import { TrackedEventData } from "@/../back-end/src/types/Integration";
+import {
+  DataSourceInterfaceWithParams,
+  DataSourceSettings,
+} from "@/../back-end/types/datasource";
+import { cloneDeep } from "lodash";
+import { useForm } from "react-hook-form";
+import { MetricType } from "@/../back-end/types/metric";
+import track from "@/services/track";
+import { useAuth } from "@/services/auth";
+import { useDefinitions } from "@/services/DefinitionsContext";
+import Button from "./Button";
+import { DocLink } from "./DocLink";
+import Modal from "./Modal";
+import Tooltip from "./Tooltip/Tooltip";
+import AutoMetricCard from "./Settings/AutoMetricCard";
+import SelectField from "./Forms/SelectField";
+import LoadingOverlay from "./LoadingOverlay";
+
+type Props = {
+  setShowAutoGenerateMetricsModal: (show: boolean) => void;
+  datasource?: DataSourceInterfaceWithParams;
+  source: string;
+  mutate: () => void;
+};
+
+export default function AutoGenerateMetricsModal({
+  setShowAutoGenerateMetricsModal,
+  datasource,
+  source,
+  mutate,
+}: Props) {
+  const [autoMetricError, setAutoMetricError] = useState("");
+  const [trackedEvents, setTrackedEvents] = useState<TrackedEventData[]>([]);
+  const { datasources } = useDefinitions();
+  const [selectedDatasource, setSelectedDatasource] = useState<
+    DataSourceInterfaceWithParams | undefined
+  >(datasource || undefined);
+  const { apiCall } = useAuth();
+  const [loading, setLoading] = useState(false);
+
+  const form = useForm<{
+    datasourceId: string;
+    settings: DataSourceSettings | undefined;
+    metricsToCreate: {
+      name: string;
+      sql: string;
+      type: MetricType;
+    }[];
+  }>({
+    defaultValues: {
+      datasourceId: datasource?.id || "",
+      settings: datasource?.settings || {},
+      metricsToCreate: [],
+    },
+  });
+
+  const submit = async () => {
+    track("Generating Auto Metrics For User", {
+      metricsToCreate: form.watch("metricsToCreate"),
```

Comment:
Sending this whole list is too invasive for telemetry data since it reveals actual event names in the user's warehouse.  We could just send the total number of metrics being created (or maybe 2 numbers - one for binomial and one for count).

Reply:
Also, don't use `form` inside the submit callback.  Instead do:

```ts
const submit = form.handleSubmit(async (data) => {
  data.metricsToCreate; // instead of form.watch("metricsToCreate")
})
```

Reply:
@jdorn I went the route of passing in two numbers, 1 for count and 1 for binomial for additional insight.

I've also updated the submit callback to not use `form` - Can you outline why you don't like to pass `form` into the callback?

Reply:
I don't 100% know how the useForm hook works under the hood, but I suspect using form within the callback could open us up to race conditions. For example, the submit callback might be fired before `form.watch` updates and you would get stale values. Not sure this ever actually happens, but the fact that the useForm authors made a `handleSubmit` method makes me think there's a good reason they didn't just tell people to access the form directly.